src/v/kafka/server/connection_context.h

```diff
@@ -283,6 +281,13 @@ class connection_context final
     using sequence_id = named_type<uint64_t, struct kafka_protocol_sequence>;
     using map_t = absl::flat_hash_map<sequence_id, response_and_resources>;
 
+    /*
+     * the top half handles the initial dispatching of the request and
+     * other consistency requirements like sequencing writes into the raft
+     * layer. the bottom half is run in the background and waits on the response
+     * to be generated, allowing processing of some requests to be overlapped.
+     */
+    ss::future<> handle_request_top_half(request_header, size_t sz);
```

Comment:
nit: hmm, maybe it would be worth to stick to the existing terminology and use `stages` or `phases` ?

Reply:
works for me.